src/lib.rs

```diff
@@ -39,22 +68,51 @@ pub trait Game: Clone + Send + Sync {
 }
 
 struct Node<G: Game> {
-    /// Move which entered this node
+    /// Move which entered this node (None if not generating IDs)
+    id: Option<String>,
     mov: Option<G::Move>,
     parent: Option<Weak<Node<G>>>,
     children: RwLock<Vec<Arc<Node<G>>>>,
     player_just_moved: Option<G::PlayerTag>,
     statistics: RwLock<NodeStatistics>,
+    generate_node_ids: bool,
```

Comment:
I don't really like saving this into every `Node`. I think checking if the parent has an `id` is equivalent